sql/core/src/main/scala/org/apache/spark/sql/execution/GeneratedAggregate.scala

```diff
@@ -265,7 +284,49 @@ case class GeneratedAggregate(
 
         val resultProjection = resultProjectionBuilder()
         Iterator(resultProjection(buffer))
+      } else if (unsafeEnabled && schemaSupportsUnsafe) {
+        log.info("Using Unsafe-based aggregator")
+        val aggregationMap = new UnsafeFixedWidthAggregationMap(
+          newAggregationBuffer(EmptyRow),
+          aggregationBufferSchema,
+          groupKeySchema,
+          SparkEnv.get.unsafeMemoryManager,
+          1024 * 16,
```

Comment:
AFAIK we can't use the Scala named argument syntax because `UnsafeFixedWidthAggregationMap` is implemented in Java, but I can add a line comment explaining this.  Also, I wonder if there's a better choice of default map size here.  If we had statistics / estimates from the optimizer, we could make a better choice here.